File: NonRepeatingCharacter.py
class Solution:
    
    #Function to find the first non-repeating character in a string.
    def nonrepeatingCharacter(self,s):
        
        #using hash table to store count of each character.
        occurences=[0 for i in range(256)]
        
        #iterating over the string.
        for char in s:
            occurences[ord(char)]+=1
        
        for i in range(len(s)):
            #if count of current character is 1, we return it. 
            if(occurences[ord(s[i])]==1):
                return s[i]
                
        #if there is no non-repeating character then we return '$'.
        return '$'
    

File: NonRepeatingCharacter.java
class Solution
{
    //Function to find the first non-repeating character in a string.
    static char nonrepeatingCharacter(String S)
    {
        //Your code here
        
        HashMap<Character,Integer> map=new HashMap<Character,Integer>();
        
        
        char ans='\0';
        //char subAns='-1';
        //String answer='\-1';
        
        //char subAns=answer.charAt(0);
        
        for(int i=0;i<S.length();i++){
            if(map.containsKey(S.charAt(i))){
                map.put(S.charAt(i),1+map.get(S.charAt(i)));
            }
            
            else{
                map.put(S.charAt(i),1);
            }
            
            
        }
        
        for(int i=0;i<S.length();i++){
            if(map.get(S.charAt(i))==1){
                ans=S.charAt(i);
                return ans;
            }
        }
        
        return '$';
    }
}
